qa/smoke-test-http/src/test/java/org/elasticsearch/http/snapshots/RestGetSnapshotsIT.java

```diff
@@ -176,7 +187,23 @@ public void testSortAndPaginateWithInProgress() throws Exception {
             inProgressSnapshots.add(AbstractSnapshotIntegTestCase.startFullSnapshot(logger, repoName, snapshotName, false));
         }
         AbstractSnapshotIntegTestCase.awaitNumberOfSnapshotsInProgress(logger, inProgressCount);
-
+        AbstractSnapshotIntegTestCase.awaitClusterState(logger, state -> {
+            boolean firstIndexSuccessfullySnapshot = state.custom(SnapshotsInProgress.TYPE, SnapshotsInProgress.EMPTY)
+                .asStream()
+                .flatMap(s -> s.shards().stream())
+                .allMatch(
+                    e -> e.getKey().getIndexName().equals("test-index-1") == false
+                        || e.getValue().state() == SnapshotsInProgress.ShardState.SUCCESS
+                );
+            boolean secondIndexIsBlocked = state.custom(SnapshotsInProgress.TYPE, SnapshotsInProgress.EMPTY)
```

Comment:
We shouldn't need this second check. If we make sure that all snapshots for index-1 are successful, then that should be the end state of things. The state of snapshots for the second index should be initialized to exactly what we're asserting here right away I think?

Reply:
https://github.com/elastic/elasticsearch/pull/80530#discussion_r758346965

@original-brownbear That was the original implementation (with only the `test-index-1` check) and you suggested to add an addition check for the second index :smiley:

Reply:
I don't have a strong feeling towards this one way or another (the test passes on 100 iterations both with `secondIndexIsBlocked` check and without it).

Reply:
Right, now I remember :) Sorry about that.